electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -279,6 +279,35 @@ def update_production_breakdowns(
 
         return production_breakdowns
 
+    @staticmethod
+    def filter_only_zero_production(
+        breakdowns: "ProductionBreakdownList",
+    ) -> "ProductionBreakdownList":
+        """A method to filter out production breakdowns with a total production of 0 MW."""
+        events = ProductionBreakdownList(breakdowns.logger)
+        for event in breakdowns.events:
+            total_production = sum(
+                {
+                    k: v for k, v in event.production.dict().items() if v is not None
+                }.values()
+            )
+            valid = True
+            if total_production == 0:
+                valid = False
+                events.logger.warning(
+                    f"Discarded production event for {event.zoneKey} at {event.datetime} because total production is 0MW."
+                )
+                continue
+            if valid:
+                events.append(
+                    zoneKey=event.zoneKey,
+                    datetime=event.datetime,
+                    production=event.production,
+                    storage=event.storage,
+                    source=event.source,
+                )
```

Comment:
```suggestion
            if event.production is not None and all(
                (v[1] == 0 or v[1] is None) for v in event.production
            ):
                events.logger.warning(
                    f"Discarded production event for {event.zoneKey} at {event.datetime} because all production values are 0."
                )
                continue
            events.append(
                zoneKey=event.zoneKey,
                datetime=event.datetime,
                production=event.production,
                storage=event.storage,
                source=event.source,
                )
```
This should be a bit simpler and easier to understand

Reply:
I agree with Viktor's suggestion, I think it's more efficient.